.github/workflows/default-tests.yml

```diff
@@ -0,0 +1,146 @@
+name: Default Testing
+
+on: 
+  pull_request:
+    branches: [dev, main, release*]
+  push:
+    branches: [dev, main, release*]
```

Comment:
Same as above

Reply:
This is actually required for codeQL to work, it need the test to be run on the branch (push) that it's merging into so that it has something to compare against when it's run on pull_request, where we need it for proper "gating" of our branches.

Here also, testing prior to merge, doesn't always mean things are guaranteed to work post merge, so having it in both helps shorten feedback loops.